synapseclient/core/credentials/credential_provider.py

```diff
@@ -104,47 +94,39 @@ def _create_synapse_credential(
 
 
 class UserArgsCredentialsProvider(SynapseCredentialsProvider):
+    def _get_auth_info(self, syn, user_login_args):
+
+        username = user_login_args.username
+        token = user_login_args.auth_token
+        # username = user_login_args.get("username")
+        # token = user_login_args.get("auth_token")
+
+        if username and token:
+            return username, token
+
+        return None, None
     """
     Retrieves auth info from user_login_args during a CLI session.
     """
 
-    def _get_auth_info(
-        self, syn: "Synapse", user_login_args: Dict[str, str]
-    ) -> Tuple[str, str]:
-        return (
-            user_login_args.username,
-            user_login_args.auth_token,
-        )
-
-
 class ConfigFileCredentialsProvider(SynapseCredentialsProvider):
     """
     Retrieves auth info from the `~/.synapseConfig` file
     """
 
-    def _get_auth_info(
-        self, syn: "Synapse", user_login_args: Dict[str, str]
-    ) -> Tuple[Union[str, None], Union[str, None]]:
-        config_dict = get_config_authentication(config_path=syn.configPath)
-        # check to make sure we didn't accidentally provide the wrong user
-
-        username = config_dict.get("username")
-        token = config_dict.get("authtoken")
-
-        if user_login_args.username and username != user_login_args.username:
-            # if the username is provided and there is a config file username but they
-            # don't match then we don't use any of the values from the config
-            # to prevent ambiguity
-            username = None
-            token = None
-            syn.logger.warning(
-                f"{user_login_args.username} was defined in the user login "
-                "arguments, however, it is also defined in the `~/.synapseConfig` "
-                "file. Becuase they do not match we will not use the `authtoken` "
-                "in the `~/.synapseConfig` file.",
-            )
+    def _get_auth_info(self, syn, user_login_args):
+        """
+    Retrieves authentication info from user_login_args during a CLI session.
+    """
+        # If authToken is explicitly provided, return it directly and skip profiles
+        if user_login_args.auth_token:
+            return None, user_login_args.auth_token  # No username needed, just authToken
 
-        return username, token
+        # Otherwise, fall back to profile-based lookup
+        if not user_login_args.profile:
+            raise SynapseAuthenticationError("No authentication method provided (neither authToken nor profile).")
+
+        raise SynapseAuthenticationError(f"Profile '{user_login_args.profile}' not found in {syn.configPath}")
```

Comment:
I'm not sure that this is correct. You have 2 branches here:

1. `if not user_login_args.profile:` -> An exception will be raised.
2. If ^ does not evaluate to True an exception is still raised.